### DISCUSSION TITLE
elastalert-server supports OpenSearch?

### DISCUSSION THREAD

**Discussion by nvasoft** [jertel/elastalert2#1404] [Q&A] [answered]
Hello,
does elastalert-server support OpenSearch?
I installed elastalert2, it works, but starting the elastalert-server via `npm run start` gives an error. The elastalert-server settings are the same as in my elasticsearch
```
12:01:10.983Z ERROR elastalert-server:
    ProcessController:  Traceback (most recent call last):
      File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
        return _run_code(code, main_globals, None,
      File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
        exec(code, run_globals)
      File "/opt/elastalert2/elastalert/create_index.py", line 18, in <module>
        from elastalert.util import get_version_from_cluster_info
      File "/opt/elastalert2/elastalert/util.py", line 47, in <module>
        def _find_es_dict_by_key(lookup_dict: dict, term: str, string_multi_field_name: str = ".keyword") -> tuple[dict, str]:
    TypeError: 'type' object is not subscriptable
```

**Comment by nsano-rururu** [comment]
not support

**Reply by nsano-rururu** [reply]
elastalert-server must not be officially published. It's not polite to ask here.

**Reply by nvasoft** [reply]
Sorry, I didn't know

**Reply by nsano-rururu** [reply, marked as answer]
@nvasoft 
It seems that there is a setting, so it may be compatible with OpenSearch.
https://github.com/johnsusek/elastalert-server
![image](https://github.com/jertel/elastalert2/assets/22293449/27c7169a-5e8c-4870-a60a-79291688fc89)

**Reply by nvasoft** [reply]
Yes. I solved the problem, Default python3.8 in ubuntu )
but another one arose. When sending an alert via SMTP, an error appears:
`Error connecting to SMTP host: EOF occurred in violation of protocol (_ssl.c:1000)`
YELP had a different approach to 'STARTTLS' in alerts.py and it worked with my settings on port 25 And elastalert 2 imports SMTP from the smtplib library.  I don't know yet how to solve the problem
P.S.
Thank you for your response

**Reply by nsano-rururu** [reply]
@nvasoft 
I don't know if I can give you advice unless I know how to configure elastalert2.

**Reply by nsano-rururu** [reply]
https://github.com/jertel/elastalert2/blob/master/elastalert/alerters/email.py#L110

**Reply by nvasoft** [reply]
My problem: the exchange-server does not send any HTTP headers (i.e. HTTP/0.9 style response)
`curl: (1) Received HTTP/0.9 when not allowed`
curl allowed HTTP/0.9 responses by default before 7.66.0. My version curl 7.68.0. How do I tell elastaler2 to accept an HTTP/0.9 response like this?

**Reply by nsano-rururu** [reply]
The problem is that since curl 7.66.0 HTTP/0.9 is disabled by default. However, this is okay because this protocol can be optionally re-enabled.

The --http0.9 option tells curl to handle HTTP version 0.9 responses normally.

**Reply by nvasoft** [reply]
That's how the mail goes
```
curl --HTTP0.9 \
  --url '192.168.2.50:25' \
  --user 'username:password' \
  --mail-from 'username@example.com' \
  --mail-rcpt 'john@example.com' \
  --upload-file mail.txt
```
If add curl --ssl, the message will not go.
If via elastalert2:
`python -m elastalert.elastalert --rule rules/DNS/01HDNS_Spyware_m.yaml --config ./config.yaml`
the error comes out:
`ERROR:elastalert:Error while running alert email: Error connecting to SMTP host: EOF occurred in violation of protocol (_ssl.c:1000)`
tspdump shows
![image](https://github.com/jertel/elastalert2/assets/99728192/733ec880-ba24-438b-8c41-f8f17fb09644)

I don't understand. Four eyes see better than two :)

**Reply by nvasoft** [reply]
An email is sent via Yelp\elastalert
![image](https://github.com/jertel/elastalert2/assets/99728192/9cf95bcb-271b-4ff2-9b52-b319a8d4d339)

**Reply by nvasoft** [reply]
I have Paton 3.8 on Ubuntu, an alternative Python 3.12. I created a simple code.
code works for me sending mail using python3.8.

in alternative python3.12 this code gives my error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/elastalert2/my_email.py", line 20, in <module>
    sendemail(from_addr    = 'MyName@xxx.com',
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/elastalert2/my_email.py", line 14, in sendemail
    server.starttls()
  File "/usr/local/lib/python3.12/smtplib.py", line 779, in starttls
    self.sock = context.wrap_socket(self.sock,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/ssl.py", line 455, in wrap_socket
    return self.sslsocket_class._create(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/ssl.py", line 1046, in _create
    self.do_handshake()
  File "/usr/local/lib/python3.12/ssl.py", line 1317, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:1000)
```
**ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:1000)**

**Comment by jertel** [comment]
The Exchange server is closing the socket. Why? Did you check the Exchange server logs?

This SMTP topic should have been a new discussion.
